experimental/linting_tool/lib/pages/home_page.dart

```diff
@@ -3,13 +3,74 @@
 // found in the LICENSE file.
 
 import 'package:flutter/material.dart';
+import 'package:linting_tool/model/rules_store.dart';
+import 'package:linting_tool/layout/adaptive.dart';
+import 'package:linting_tool/widgets/lint_expansion_tile.dart';
+import 'package:provider/provider.dart';
 
-class HomePage extends StatelessWidget {
+class HomePage extends StatefulWidget {
   const HomePage({Key? key}) : super(key: key);
 
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance!.addPostFrameCallback((timeStamp) {
+      var rulesStore = Provider.of<RuleStore>(context, listen: false);
+      if (rulesStore.rules.isEmpty && !rulesStore.isLoading) {
+        rulesStore.fetchRules();
```

Comment:
I'm curious why we have this `rulesStore.fetchRules()` load logic here specifically. It feels like something we should farm off to an isolate so we don't lag the main thread if CPU is a concern.